Utilities/Thread.cpp

```diff
@@ -1905,6 +1905,11 @@ void thread_base::start()
 	m_thread = ::_beginthreadex(nullptr, 0, entry_point, this, CREATE_SUSPENDED, nullptr);
 	ensure(m_thread);
 	ensure(::ResumeThread(reinterpret_cast<HANDLE>(+m_thread)) != -1);
+#elif defined(__APPLE__)
+	pthread_attr_t stack_size_attr;
+	pthread_attr_init(&stack_size_attr);
+	pthread_attr_setstacksize(&stack_size_attr, 0x200000);
```

Comment:
Don't we just want `max(current_stack_size, 2MB)`?
What happens if OS stack size is greater than 2MB here?

Reply:
May be useful to use 8 M for consistency with other platforms.

Reply:
This is secondary thread stack size, not main. We believe that it currently defaults to 1 mb on windows and 2 mb on Linux, so this change would match the Linux behavior. Mac default is 512 kb which was causing crashes.

Reply:
Stack size on Windows is configured to 8 MiB because otherwise it runs into the same issue (stack overflow in SPU LLVM).

Reply:
I think thread stack size is 8 MiB by default on 64-bit Linux, too.
I'll change it on macOS accordingly.